src/platform/platform_worker.c

```diff
@@ -300,6 +307,65 @@ CxPlatWorkersUninit(
     CxPlatLockUninitialize(&CxPlatWorkerLock);
 }
 
+#define DYNAMIC_POOL_PROCESSING_TIME 1000000 // 1 second
+#define DYNAMIC_POOL_PRUNE_COUNT     8
+
+void
+CxPlatAddDynamicPoolAllocator(
+    _Inout_ CXPLAT_POOL_EX* Pool,
+    _In_ uint16_t Index // Into the execution config processor array
+    )
+{
+    CXPLAT_WORKER* Worker = &CxPlatWorkers[Index];
+    Pool->Owner = Worker;
+    CxPlatLockAcquire(&Worker->ECLock);
+    CxPlatListInsertTail(&CxPlatWorkers[Index].DynamicPoolList, &Pool->Link);
+    CxPlatLockRelease(&Worker->ECLock);
+}
+
+void
+CxPlatRemoveDynamicPoolAllocator(
+    _Inout_ CXPLAT_POOL_EX* Pool
+    )
+{
+    CXPLAT_WORKER* Worker = (CXPLAT_WORKER*)Pool->Owner;
+    CxPlatLockAcquire(&Worker->ECLock);
+    CxPlatListEntryRemove(&Pool->Link);
+    CxPlatLockRelease(&Worker->ECLock);
+}
+
+void
+CxPlatProcessDynamicPoolAllocator(
+    _Inout_ CXPLAT_POOL_EX* Pool
+    )
+{
+    for (uint32_t i = 0; i < DYNAMIC_POOL_PRUNE_COUNT; ++i) {
+        void* Entry = InterlockedPopEntrySList(&Pool->ListHead);
+        if (!Entry) break;
+        Pool->Free(Entry, Pool->Tag, (CXPLAT_POOL*)Pool);
+    }
+}
+
+void
+CxPlatProcessDynamicPoolAllocators(
+    _In_ CXPLAT_WORKER* Worker
+    )
+{
+    QuicTraceLogVerbose(
+        PlatformWorkerProcessPools,
+        "[ lib][%p] Processing pools",
+        Worker);
+
+    CxPlatLockAcquire(&Worker->ECLock);
```

Comment:
Is this lock heavily contended? If the worker has to scan 1000+ CPUs and free 8 items per list per CPU, that could add some jitter.

Reply:
Ah, this is a per-worker lock, so you're only looking at 8 items per list. Seems pretty lightweight.